library/core/src/main/java/com/google/android/exoplayer2/mediacodec/MediaCodecRenderer.java

```diff
@@ -768,7 +768,11 @@ private void maybeInitCodecWithFallback(
         return;
       }
       try {
-        initCodec(codecInfo, crypto);
+        if (codecInfo != null && !codecInfo.isCodecSupported(inputFormat.codecs)) {
+            availableCodecInfos.removeFirst();
+        }else {
+            initCodec(codecInfo, crypto);
```

Comment:
This is just to select the codec with the right profile. 
I am almost sure another method exists.

Reply:
The expected flow is that `MediaCodecVideoRenderer.supportsFormat` gets called, we pick the first DV decoder, then we check if its profile/level support includes the input media. If so, then we know we can play the input media so the renderer will be used and we'll initialize the first DV decoder. Picking a non-primary DV decoder (further down the list) can wait for a later change.